### ISSUE TITLE
TypeError: Cannot read property 'identify' of undefined

### ISSUE THREAD

**Issue by jhunflx** [poteto/ember-metrics#79]
Hello,

I'm getting the "Cannot read property" error when I try to run the following code:
```
    @get('metrics').identify('Mixpanel', {
      distinctId: 'username@email.com'
    })
```

The error is coming from metrics.js line 130:
https://github.com/poteto/ember-metrics/blob/master/addon/services/metrics.js#L130

This code has been running fine in my 'production' environment.  When I recently added a 'localhost' environment is when I got the error.

My config/environment.js looks like this:
```
  if (environment === 'localhost') {
      ENV.apiHost = 'http://localhost';
      ENV.apiURL = 'http://localhost'
      ENV.metricsAdapters = [{
          name: 'Mixpanel',
          environments: ['development'],
          config: { token: 'xxxxx' },
          includeAdapters: ['mixpanel']
      }, ]
  }
  if (environment === 'production') {
    ENV.apiHost = 'https://my-domain.com';
    ENV.apiURL = 'https://my-domain.com';
    ENV.metricsAdapters = [
      {
        name: 'Mixpanel',
        environments: ['production'],
        config:{ token: 'xxxxx' },
        includeAdapters: ['mixpanel']
      },
    ]
  }
```

Is there something weird because I have this 'localhost' env?

Any help is appreciated!  Thanks!

**Comment by poteto** [comment]
Hi, sorry about the delay!

It looks like your `config/environment.js` might be slightly misconfigured: You could try:

```js
module.exports = function(environment) {
  var ENV = {
    metricsAdapters: [
      {
        name: 'Mixpanel',
        environments: ['development'],
        config: {
          token: 'devtoken'
        }
      },
      {
        name: 'Mixpanel',
        environments: ['production'],
        config: {
          token: 'prod_token'
        }
      }
    ]
  }
}
```

**Comment by jhunflx** [comment]
Hi!  Thanks for the reply!

Just curious, in your documentation, you had mentioned that the valid environments are `development`, `test`,`production`,`all`.  I have custom environments in my project like `localhost`, `staging` and `demo`.  Does this mean that ember-metrics will not work on those environments?

I changed the config to your suggestion and I'm still getting the same error.  I'm pretty sure it's because of my custom environments, but I wanted to make sure this is really the case.  This addon works fine in our actual `development` and `production` environments.

Thanks again!

Julie

**Comment by poteto** [comment]
Hey Julie, yeah those are the default environments, but they're not set in stone. The metrics-service will basically lookup your current environment and compare that with the one it finds in the metrics config. Could you try adding `localhost ` to the `environments` array for the first adapter in the config?

**Comment by jhunflx** [comment]
Ha...that did the trick!

Thank you so much for your help, Lauren!

**Comment by poteto** [comment]
No problem! 😄

**Comment by XaserAcheron** [comment]
My apologies for 'bumping' a closed issue, but I've run into a variant of this. The short version is that I'm trying to _disable_ ember-metrics in the `development` environment. If I set the environments array to just `['production']`, `identify` blows up as described when running in `development`.

Is there a more explicit way of saying "turn it off"? Right now I've got a custom check in place before calling `identify` but it seems unnecessary -- i.e. can we have it so calls to `identify` ought do nothing if ember-metrics is disabled for the current environment?

`trackPage` doesn't exhibit the same issue, for the record. Haven't tested the other two since I'm just starting out with the plugin.
